### PR TITLE
Removed static activity source from hosting

### LINKED ISSUE #30582: Test Failure: UseStartupFactoryWorks
Build: https://dev.azure.com/dnceng/public/_build/results?buildId=1017405&view=ms.vss-test-web.build-test-results-tab&runId=31669222&resultId=101504&paneView=debug

Log:
```
Assert.Equal() Failure
↓ (pos 1)
Expected: 0123456789abcdef
Actual:   0000000000000000
↑ (pos 1)
   at Microsoft.AspNetCore.Hosting.Tests.HostingApplicationDiagnosticsTests.<>c.<ActivityListenersAreCalled>b__17_2(Activity activity) in /_/src/Hosting/Hosting/test/HostingApplicationDiagnosticsTests.cs:line 506
   at System.Diagnostics.ActivitySource.<>c.<NotifyActivityStart>b__24_0(ActivityListener listener, Object obj)
   at System.Diagnostics.SynchronizedList`1.EnumWithAction(Action`2 action, Object arg)
   at System.Diagnostics.ActivitySource.NotifyActivityStart(Activity activity)
   at System.Diagnostics.Activity.Start()
   at Microsoft.AspNetCore.Hosting.HostingApplicationDiagnostics.StartActivity(HttpContext httpContext, Boolean loggingEnabled, Boolean diagnosticListenerActivityCreationEnabled, Boolean& hasDiagnosticListener) in /_/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs:line 309
   at Microsoft.AspNetCore.Hosting.HostingApplication.CreateContext(IFeatureCollection contextFeatures) in /_/src/Hosting/Hosting/src/Internal/HostingApplication.cs:line 80
   at Microsoft.AspNetCore.Hosting.WebHostBuilderTests.TestServer.<>c__DisplayClass8_0`1.<<StartAsync>b__0>d.MoveNext()
--- End of stack trace from previous location ---
   at Microsoft.AspNetCore.Hosting.WebHostBuilderTests.AssertResponseContains(RequestDelegate app, String expectedText) in /_/src/Hosting/Hosting/test/WebHostBuilderTests.cs:line 1503
   at Microsoft.AspNetCore.Hosting.WebHostBuilderTests.UseStartupFactoryWorks(IWebHostBuilder builder) in /_/src/Hosting/Hosting/test/WebHostBuilderTests.cs:line 132
--- End of stack trace from previous location ---
```

### PR DESCRIPTION
**Pull request by davidfowl**
- Get it from DI instead

Fixes https://github.com/dotnet/aspnetcore/issues/30582

### CODE REVIEW

File: src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs [hunk 1]
```diff
@@ -86,6 +86,7 @@ public GenericWebHostBuilder(IHostBuilder builder, WebHostBuilderOptions options
                 // We need to flow this differently
                 services.TryAddSingleton(sp => new DiagnosticListener("Microsoft.AspNetCore"));
                 services.TryAddSingleton<DiagnosticSource>(sp => sp.GetRequiredService<DiagnosticListener>());
+                services.TryAddSingleton(sp => new ActivitySource("Microsoft.AspNetCore"));
```
Comment (Tratcher) [inline src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs:89]: Why did the value change from "Microsoft.AspNetCore.Hosting"?

**Review by Tratcher — APPROVED** [review]

File: src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs (hunk 1 above)
Comment (davidfowl) [reply src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs:89]: Matches the diagnostic listener. @shirhatti are they used in a similar way?

Comment (shirhatti) [reply src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs:89]: This is technically a breaking change. You've changed the ActivitySource name. Most subscribers only care about the Activity name, but I'd prefer not to have this change

Comment (davidfowl) [reply src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs:89]: Didn't we make this change in 6.0?

Comment (shirhatti) [reply src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs:89]: The original ActivitySource guidance was to have something like ActivitySource Name = "Microsoft.AspNetCore.Hosting" and Activity Name = "BeginRequest", that is don't use a fully qualified name for the Activity. However, it wasn't possible to do that without breaking existing consumers, so we just changed the guidance for Activity 😄

Comment (shirhatti) [reply src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs:89]: Yes, new in 6.0

**Comment by shirhatti** [comment]
Thanks for fixing that test, it was on my backlog 😄

**Review by shirhatti — APPROVED** [review]

File: src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs (hunk 1 above)
Comment (davidfowl) [reply src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs:89]: Then it's not a breaking change. We just need the "right" naming.

**Pull request merged**